app/controllers/posts_controller.rb

```diff
@@ -16,10 +19,26 @@ def new
   end
 
   def create
-    @user = current_user
-    @post = Post.new(author_id: @user, title: params[:post][:title], text: params[:post][:text])
-    @post.author_id = @user.id
-    @post.save
-    redirect_to user_posts_path(@user)
+    @post = Post.new(post_params)
+    @post.author = current_user
+    @post.likes_counter = 0
+    @post.comments_counter = 0
+    return unless @post.save
+
+    redirect_to user_posts_path
+    flash[:success] = 'Post created!'
+  end
+
+  def destroy
+    @post = current_user.posts.find(params[:id])
+    @post.destroy
+    flash[:success] = 'Post deleted!'
+    redirect_to user_posts_path
+  end
+
```

Comment:
- [ ] The delete post functionality does not work for the admin role. Only the post author can delete a post. Kindly ensure the admin user can delete a post. Right now the admin can only delete a comment.